core/src/main/java/io/questdb/griffin/SqlCodeGenerator.java

```diff
@@ -2337,6 +2398,34 @@ private RecordCursorFactory generateSelectVirtual(QueryModel model, SqlExecution
         }
     }
 
+    private static boolean builtInFunctionCast(int fromColumnType, int toColumnType) {
```

Comment:
It is a subset of `SqlCompiler.isAssignableFrom`, only returns of types which don't need explicit cast from one value to another because for example `IntFunction` has `getDouble` implemented

Reply:
ok, moved `builtInFunctionCast` to `SqlCompiler` and refactored `isAssignableFrom` to use it as one of the parts